### PR TITLE
feat(#224): agentic answers + pre visit inquiry implemented

### PR DESCRIPTION
**Pull request by pkarw**

### CODE REVIEW

File: src/components/chat-message.tsx [hunk 1]
```diff
@@ -90,11 +93,17 @@ const ChatMessage: React.FC<ChatMessageProps> = ({ message, ref }) => {
                     </AccordionItem>
                 </Accordion>
               </div>
-            ) : (
+            ) : ((message.displayMode === 'jsonAgentResponse' ? (              
+              (!message.finished ? <>Thinking ...</>  : (
+                <Markdown className={styles.markdown} remarkPlugins={[remarkGfm]}>
```
Comment (Bukareszt) [inline src/components/chat-message.tsx:98]: More readable: 
  {message.messageAction?.type === 'agentQuestion'
    ? removeCodeBlocks(message.content) + message.messageAction.params.question
    : removeCodeBlocks(message.content)}

File: src/contexts/chat-context.tsx [hunk 2]
```diff
@@ -290,6 +366,32 @@ export const ChatContextProvider: React.FC<PropsWithChildren> = ({ children }) =
         }, providerName, modelName); // TODO: add an option to auto check with different models
     }
 
+    /// TODO: instead of actions we could process tool calls but actually it didn't matter that much
+    const processMessageAction = (jsonObject: MessageAction, resultMessage: MessageEx) => {
+        console.log(jsonObject);
```
[the remark this answers is not in the document]
Comment (pkarw) [reply src/contexts/chat-context.tsx]: fixed thanks!

File: src/components/chat-message.tsx (hunk 1 above)
Comment (pkarw) [reply src/components/chat-message.tsx]: Yes, but I've added this "Thinking ..." message which is out of the <Markdown/> so I guess it should stay as it is (?)

**Review by Bukareszt — APPROVED** [review]
Approve - I am not yet familiar with code bass to be able to suggest a different approach

**Pull request merged**